pkg/core/transaction/transaction.go

```diff
@@ -329,6 +334,9 @@ func (t *Transaction) UnmarshalJSON(data []byte) error {
 	if t.Hash() != tx.TxID {
 		return errors.New("txid doesn't match transaction hash")
 	}
+	if t.Size() != tx.Size {
+		return errors.New("'size' doesn't match transaction size")
+	}
```

Comment:
Wouldn't it be better to put these two checks inside of `(t *Transaction) isValid()`? `isValid()` method is aimed to unify transaction decoding checks, it is called from both `(t *Transaction) UnmarshalJSON` and `(t *Transaction) decodeHashableFields`. In this case we also have to move `t.isValid()` call from `decodeHashableFields` to `DecodeBinary`.

Reply:
These are JSON-specific, you just don't have them in the binary encoding.